projects/igniteui-angular/src/lib/test-utils/grid-samples.spec.ts

```diff
@@ -1591,13 +1591,32 @@ export class DynamicColumnsComponent extends GridWithSizeComponent {
             <igx-checkbox [checked]="headContext.totalCount === headContext.selectedCount"
                 (click)="onHeaderCheckboxClick($event, headContext)"></igx-checkbox>
         </ng-template>
-    </igx-grid>`
+    </igx-grid>
+    <ng-template #customRow igxRowSelector let-rowContext>
+            <span class="rowNumber">CUSTOM SELECTOR: {{ rowContext.index }}</span>
+    </ng-template>
+    <ng-template #customHeader igxHeadSelector let-headContext>
+        <span>CUSTOM HEADER SELECTOR</span>
+    </ng-template>
+    <ng-template #customGroupRow igxHeadSelector let-headContext>
+        <span>CUSTOM GROUP SELECTOR</span>
+    </ng-template>
+    `
 })
 export class GridCustomSelectorsComponent extends BasicGridComponent implements OnInit {
     @ViewChild('gridCustomSelectors', { static: true })
     public grid: IgxGridComponent;
     public rowCheckboxClick: any;
     public headerCheckboxClick: any;
+    @ViewChild('customRow', {read: TemplateRef, static: true })
+    public customRowTemplate: TemplateRef<any>;
+
+    @ViewChild('customHeader', {read: TemplateRef, static: true })
```

Comment:
Also, do we need these viewchilds because there are not tests that use these variables?

Reply:
@mddragnev 
We use them in the newly added test to set the grid templates: https://github.com/IgniteUI/igniteui-angular/pull/12199/files#diff-e484c9e552a05bc63fa13e22b7f3dade2ea58bb27b31992d3dee7e22a996b583R2198